### PR TITLE
Feat: Extending report interaction with static context that can be appended to all interaction events

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What is this feature?**
This PR introduces a new config property called `analytics.reportingStaticContext` which defines key=value pairs that will be attached to all `reportInteraction` events. The attached properties are always prefixed with `_static_context_`.

Here is how the event properties look like:

**before:**
SCR-20240612-pqbu

**after:**
SCR-20240612-pqco

**Why do we need this feature?**
Making it easier to extend events with generic data that is relevant for all interactions. 

**Who is this feature for?**
Mostly those who use analytics extensively and have a fleet of instances to monitor and distinguish

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by tolzhabayev** [comment]
Ping @jackw

File: packages/grafana-runtime/src/analytics/utils.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { isObject } from 'lodash';
```
[the remark this answers is not in the document]
Comment (tolzhabayev) [reply packages/grafana-runtime/src/analytics/utils.ts]: Good point, changed https://github.com/grafana/grafana/pull/88927/commits/4fbd48a7a30c5e5459b01980bc435dd15d1181fb

File: packages/grafana-runtime/src/config.ts [hunk 2]
```diff
@@ -221,6 +222,20 @@ export class GrafanaBootConfig implements GrafanaConfig {
...
+      pairs.forEach((pair) => {
+        // Split each pair by the equals sign
+        const [key, value] = pair.split('=').map((str) => str.trim());
+        if (key && value) {
+          result[`_static_context_${key}`] = value;
+        }
+        this.reportingStaticContext = result;
+      });
+    }
+
```
[the remark this answers is not in the document]
Comment (tolzhabayev) [reply packages/grafana-runtime/src/config.ts]: Ok moved into backend https://github.com/grafana/grafana/pull/88927/commits/17d69fbc664ad55bea70e0ace9ba2250ab97c3a2

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Review by jackw — APPROVED** [review]
LGTM! 🚀

**Pull request merged**